libbeat/logp/logp.go

```diff
@@ -141,3 +156,45 @@ func getLogLevel(config *Logging) (Priority, error) {
 	}
 	return level, nil
 }
+
+func snapshotExpvars(varsMap map[string]int64) {
+	expvar.Do(func(kv expvar.KeyValue) {
+		switch kv.Value.(type) {
+		case *expvar.Int:
+			varsMap[kv.Key], _ = strconv.ParseInt(kv.Value.String(), 10, 64)
+		}
+	})
+}
+
+func logExpvars(metricsCfg *LoggingMetricsConfig) {
+	if metricsCfg.Enabled != nil && *metricsCfg.Enabled == false {
+		Info("Metrics logging disabled")
+		return
+	}
+	if metricsCfg.Period == nil {
+		metricsCfg.Period = &defaultMetricsPeriod
+	}
+	Info("Metrics logging every %s", metricsCfg.Period)
+
+	ticker := time.NewTicker(*metricsCfg.Period)
+	prevVals := map[string]int64{}
+	for {
+		<-ticker.C
```

Comment:
A simple version could be just passing the done channel to the go routine and using a select statement. But that would require to move the start of the goroutine into the beat starting process.